test/CMakeLists.txt

```diff
@@ -1,14 +1,16 @@
-cmake_policy(SET CMP0135 NEW)
+find_package(GTest)
 
-include(FetchContent)
-include(GoogleTest)
+if(NOT GTest_FOUND)
+    if(POLICY CMP0135)
+        cmake_policy(SET CMP0135 NEW)
+    endif()
 
-set(BUILD_GMOCK OFF)
-set(INSTALL_GTEST OFF)
-set(gtest_force_shared_crt ON)
+    include(FetchContent)
+
+    set(BUILD_GMOCK OFF)
+    set(INSTALL_GTEST OFF)
+    set(gtest_force_shared_crt ON)
```

Comment:
Why is this variable set? If gtest is not installed, does it make sense to explicitly use shared runtime? Haven't found why it is useful.

Reply:
Unless we set `gtest_force_shared_crt` to `ON`, GoogleTest's CMake proejct will override compiler options that cause an linker error on Windows. Please see [googletest/README.md](https://github.com/google/googletest/blob/main/googletest/README.md#visual-studio-dynamic-vs-static-runtimes) and [googletest/internal_utils.cmake](https://github.com/google/googletest/blob/f063cd25c90cbd4089a0ff96f5991df4f2721338/googletest/cmake/internal_utils.cmake#L34-L47).

Reply:
Okay, it might be set only for windows in that case. Or at least had comment referencing this is MSVC and Windows specific issue. ``if (MSVC) ...`` should be used if it does not make sense for any other platform. But not important for this PR.